authorization/storage_authorization.go

```diff
@@ -374,14 +602,24 @@ func authorizationsPredicateFn(f influxdb.AuthorizationFilter) kv.CursorPredicat
 	}
 
 	if f.Token != nil {
-		exp := *f.Token
+		token := *f.Token
+		allHashes, err := s.hasher.AllHashes(token)
+		if err != nil {
+			s.log.Error("error generating hashes in authorizationsPredicateFn", zap.Error(err))
+			// On error, continue onward. allHashes is empty and we'll effectively ignore hashedToken,
+			// but we'll still look at the unhashed Token if it is available.
+		}
 		return func(_, value []byte) bool {
 			// it is assumed that token never has escaped string data
-			got, _, _, err := jsonparser.Get(value, "token")
-			if err != nil {
-				return true
+			if got, _, _, err := jsonparser.Get(value, "token"); err == nil {
+				return string(got) == token
+			}
+			if len(allHashes) > 0 {
+				if got, _, _, err := jsonparser.Get(value, "hashedToken"); err == nil {
```

Comment:
Should you check if `err` is not nil? It appears that this method will throw errors for a few different [reasons](https://github.com/buger/jsonparser/blob/df3ea76ece10095374fd1c9a22a4fb85a44efc42/parser.go#L936-L966). It may be good to add a safety check if err != nil.